Add validation for length of database name

SachinTS opened the pull request:
Right now terraform provider does not have any restriction on length of the database name.
But the redislabs api limits length of the database name to max of 40 characters. This causes `terraform apply` to fail.
This commit adds validation for the same.

Error observed (only when `TF_LOG=trace` is set):
```
"additionalInfo" : "Database name is longer than 40 characters"
```

Ref issue: https://github.com/RedisLabs/terraform-provider-rediscloud/issues/99

Commit:
Add validation for length of database name

Right now terraform provider does not have any restriction on length of the database name.
But the redislabs api limits length of the database name to max of 40 characters. This causes `terraform apply` to fail.
This commit adds validation for the same.

Error observed (only when `TF_LOG=trace` is set):
```
"additionalInfo" : "Database name is longer than 40 characters"
```

Commit:
Add validation for length of database name

Right now terraform provider does not have any restriction on length of the database name.
But the redislabs api limits length of the database name to max of 40 characters. This causes `terraform apply` to fail.
This commit adds validation for the same.

Error observed (only when `TF_LOG=trace` is set):
```
"additionalInfo" : "Database name is longer than 40 characters"
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/RedisLabs/terraform-provider-rediscloud?pullRequest=100) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/RedisLabs/terraform-provider-rediscloud?pullRequest=100) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: SachinTS<br/>:x: Sachin Kumar<br/><hr/>**Sachin Kumar** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RedisLabs/terraform-provider-rediscloud?pullRequest=100) it.</sub>

SachinTS committed:
Add validation for length of database name

Right now terraform provider does not have any restriction on length of the database name.
But the redislabs api limits length of the database name to max of 40 characters. This causes `terraform apply` to fail.
This commit adds validation for the same.

Error observed (only when `TF_LOG=trace` is set):
```
"additionalInfo" : "Database name is longer than 40 characters"
```

SachinTS committed:
Merge branch 'main' of github.com:SachinTS/terraform-provider-rediscloud into main

SachinTS committed:
Merge branch 'main' of github.com:SachinTS/terraform-provider-rediscloud into main

SachinTS committed:
Merge branch 'main' of github.com:SachinTS/terraform-provider-rediscloud into main

SachinTS committed:
Add validation for length of database name

Right now terraform provider does not have any restriction on length of the database name.
But the redislabs api limits length of the database name to max of 40 characters. This causes `terraform apply` to fail.
This commit adds validation for the same.

Error observed (only when `TF_LOG=trace` is set):
```
"additionalInfo" : "Database name is longer than 40 characters"
```

SachinTS committed:
Merge branch 'main' of github.com:SachinTS/terraform-provider-rediscloud into main
